zaza/charm_lifecycle/utils.py

```diff
@@ -466,6 +466,38 @@ def is_config_deploy_forced_for_bundle(
     return False
 
 
+def is_config_deploy_trusted_for_bundle(
+        bundle_name, yaml_file=None, fatal=True):
+    """Ask the config if the bundle_name should be deployed with --trust.
+
+    The tests_options section needs to look like:
+
+        tests_options:
+          trust:
+            - focal-ussuri
+
+    e.g. trust is a list of bundle names; if the bundle is mentioned
+    then that bundle will be deployed with --trust.
+
+    :param bundle_name: the bundle to check in the trust list
+    :type bundle_name: str
+    :param yaml_file: the YAML file that contains the tests specification
+    :type yaml_file: Optional[str]
+    :param fatal: whether any errors cause an exception or are just logged.
+    :type fatal: bool
+    :returns: True if the config option is set for the bundle
+    :rtype: bool
+    :raises: OSError if the YAML file doesn't exist and fatal=True
+    """
+    config = get_charm_config(yaml_file, fatal)
+    try:
+        return bundle_name in config['tests_options']['trust']
+    # Type error is if the trust is present, but with no list
+    except (KeyError, TypeError):
+        pass
+    return False
```

Comment:
This is essentially identical to the function above except for the key `trust`.  Did this not trigger your DRY radar? :)